src/bootstrap/compile.rs

```diff
@@ -640,6 +640,142 @@ impl Step for RustcLink {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct CodegenBackend {
+    pub target: TargetSelection,
+    pub compiler: Compiler,
+    pub backend: Interned<String>,
+}
+
+impl Step for CodegenBackend {
+    type Output = ();
+    const ONLY_HOSTS: bool = true;
+    const DEFAULT: bool = true;
```

Comment:
I think we'll want this to be false at least initially, I wouldn't expect us to be building codegen backends other than llvm on CI yet? I think we had discussed `cargo check` for cranelift but not more than that.

Reply:
It didn't run on CI for x86_64-gnu-llvm-8 though.

Reply:
Hm, I am unsure that I follow you -- the check step isn't intended to run on the x86_64-gnu-llvm-8 builder, it gets run as part of mingw-check (and indeed is failing right now due to the directory not existing).

Reply:
A regular `./x.py build` happens on x86_64-gnu-llvm-8. That builder didn't build cg_clif, so cg_clif must be disabled by default when using `./x.py build`. I also think I know why. The `codegen-backends` config doesn't include `cranelift`, so `make_run` doesn't call `builder.ensure(CodegenBackend { /* ... */ })`.

Reply:
I thought we had discussed and agreed that there's no need just yet to build (not check) cranelift on CI?

Reply:
Correct, but because `codegen-backends` doesn't contain `cranelift` by default, it isn't build on CI.

Reply:
Oh, sorry, I see what you mean now. Could you add a comment on the = true to note that the default is basically conditional here?